src/row.jl

```diff
@@ -195,3 +195,176 @@ function _inputs_to_observation_table(; predicted_hard_labels::AbstractVector,
+                                                               Vector{Float64}}},
+                                        spearman_correlation::Union{Missing,
+                                                                    NamedTuple{(:ρ, :n,
+                                                                                :ci_lower,
+                                                                                :ci_upper),
+                                                                               Tuple{Float64,
+                                                                                     Int64,
+                                                                                     Float64,
+                                                                                     Float64}}},
+                                        reliability_calibration_curve::Union{Missing,
+                                                                             Tuple{Vector{Float64},
+                                                                                   Vector{Float64}}},
+                                        reliability_calibration_score::Union{Missing,
+                                                                             Float64})
+
+A type alias for [`Legolas.Row{typeof(Legolas.Schema("tradeoff-metrics@1"))}`](https://beacon-biosignals.github.io/Legolas.jl/stable/#Legolas.@row)
+representing metrics calculated over predicted soft labels.
+See also [`get_tradeoff_metrics`](@ref).
+"""
+const TradeoffMetricsRow = Legolas.@row("lighthouse.tradeoff-metrics@1" >
+                                        "lighthouse.class@1",
+                                        roc_curve::Tuple{Vector{Float64}, Vector{Float64}},
+                                        roc_auc::Float64,
+                                        pr_curve::Tuple{Vector{Float64}, Vector{Float64}},
+                                        spearman_correlation::Union{Missing,
+                                                                    NamedTuple{(:ρ, :n,
+                                                                                :ci_lower,
+                                                                                :ci_upper),
+                                                                               Tuple{Float64,
+                                                                                     Int64,
+                                                                                     Float64,
+                                                                                     Float64}}},
+                                        reliability_calibration_curve::Union{Missing,
+                                                                             Tuple{Vector{Float64},
+                                                                                   Vector{Float64}}},
```

Comment:
Another idea for later: we could define something like
```julia
struct Curve
    x::Vector{Float64}
    y::Vector{Float64}
    function Curve(x::Vector{Float64}, y::Vector{Float64})
        length(x) == length(y) || throw(DimensionMismatch("Arguments to `Curve` must have same length. Got `length(x)=$(length(x))` and `length(y)=$(length(y))`"))
        new(x,y)
    end
end
floatify(x) = convert(Vector{Float64}, replace(x, missing => NaN))
Curve(x, y) = Curve(floatify(x), floatify(y))
Curve(c::Curve) = c
Base.iterate(c::Curve, st=1) = st <= fieldcount(Curve) ? (getfield(c, st), st+1) : nothing
Base.length(::Curve) = fieldcount(Curve)
Base.getindex(c::Curve, i::Int) = getfield(c, i)
# TODO: define `==` and `hash`
```
Then this acts a lot like a `Tuple{Vector{Float64}, Vector{Float64}}`, in that you can index into it, like `C[1]`, or iterate it, `x, y = C`, but it can help handle conversion to NaNs and check dimensions. Might make some of these schemas a bit easier to read as well. And we can put in our `@row` conversions to `Curve` automatically.

Reply:
Right, added!